pyiron_atomistics/lammps/structure.py

```diff
@@ -424,11 +462,8 @@ def structure_full(self):
             neighbors = self._structure.get_neighbors_by_distance(
                 cutoff_radius=max_cutoff
             )
-            id_mol = 0
-            indices = self._structure.indices
-            for id_el, id_species in enumerate(indices):
-                id_mol += 1
-                molecule_lst.append([id_el, id_mol, species_translate_list[id_species]])
```

Comment:
Here the molecule ID was modified in the previous version but I have a hard time to see this part in your modified version.

Reply:
The previous implementation assigns a new molecule ID for each atom. While I agree that setting it to 1 is wrong, assigning the same mol and atom ids seems rather pointless also.
However, structure_bond sets it to 1, so my change at least leads to consistent behavior between the two methods.

I have to admit that I normally do not use pair styles needing molecule IDs, but I think there is no way to automatically set them correctly. IMO this could be solved via the tag system, as for example done for selective dynamics (see my previous comment in the thread). If wanted, I could implement that I guess, but since I don't really use it someone would need to test it.

Reply:
> I have to admit that I normally do not use pair styles needing molecule IDs, but I think there is no way to automatically set them correctly. IMO this could be solved via the tag system, as for example done for selective dynamics (see my previous comment in the thread). If wanted, I could implement that I guess, but since I don't really use it someone would need to test it.

I looked at other codes like ASE https://gitlab.com/ase/ase/-/blob/master/ase/io/lammpsdata.py and they also set the molecule Id to 0 if no molecule is defined. So most likely we are fine. @ahmedabdelkawy Do you know anybody using LAMMPS with molecule IDs? I remember Sudarsan used it for his water simulation at some point, but I am not sure what the status is.

Reply:
I added a possibility to manually set molecule ids via the tag system

Reply:
No, I am not familiar with the reason why it is implemented this way.